src/bp/core/messaging/messaging-router.ts

```diff
@@ -0,0 +1,38 @@
+import * as sdk from 'botpress/sdk'
+import { UnauthorizedError } from 'common/http'
+import { HTTPServer } from 'core/app/server'
+import { CustomRouter } from 'core/routers/customRouter'
+import { Router } from 'express'
+import { MessagingLegacy } from './legacy'
+import { MessagingService } from './messaging-service'
+
+export class MessagingRouter extends CustomRouter {
+  private legacy: MessagingLegacy
+
+  constructor(private logger: sdk.Logger, private messaging: MessagingService, private http: HTTPServer) {
+    super('Messaging', logger, Router({ mergeParams: true }))
+    this.legacy = new MessagingLegacy(logger, http)
+  }
+
+  public setupRoutes(): void {
+    this.router.post('/receive', async (req, res, next) => {
+      if (req.headers.password !== process.INTERNAL_PASSWORD) {
+        return next(new UnauthorizedError('Password is missing or invalid'))
+      }
+
+      const msg = req.body
+
```

Comment:
I know this is supposed to be an internal route, but the internal password is temporary, so please add a validation schema for the message. 

If you post an invalid payload to that endpoint, it crashes the web server

![image](https://user-images.githubusercontent.com/42552874/127008074-e521428b-2a7f-4aa1-b9a0-0882337cec11.png)